components/dash-html-components/scripts/generate-components.js

```diff
@@ -263,14 +269,17 @@ const ${Component} = (props) => {
     if(props.loading_state && props.loading_state.is_loading) {
         dataAttributes['data-dash-is-loading'] = true;
     }
-
+    /* remove unnecessary onClick event listeners  */
+    const isStatic = props.disable_n_clicks || typeof props.id === "undefined";
```

Comment:
```suggestion
    const isStatic = props.disable_n_clicks || !props.id;
```
NBD but we don't allow `id=""` and as a result in dash-renderer we just check if `id` is truthy for this kind of situation.